### PR TITLE
GEODE-5943: Refactor EvictionDUnitTest and OffHeapEvictionDUnitTest

### PR DESCRIPTION
**Pull request by jinmeiliao**
* break the test into dunit and integration tests
* use the rules to better see what's the setup and what the test is doing

No functional changes to the tests. This to prepare debugging the test in Jdk11. This way, we can better reasoning what's happening in the test.

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-core/src/integrationTest/java/org/apache/geode/internal/cache/eviction/EvictionIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+                    EvictionAction.LOCAL_DESTROY)),
+        f -> f.setRedundantCopies(1).setTotalNumBuckets(5));
+
+    for (int counter = 1; counter <= maxEntries + extraEntries; counter++) {
+      pr1.put(counter, new byte[1024 * 1024]);
+    }
+
+    assertThat(pr1.getTotalEvictions()).isEqualTo(extraEntries);
+
+    for (final Iterator i = pr1.getDataStore().getAllLocalBuckets().iterator(); i.hasNext();) {
```
Comment (PurelyApplied) [inline geode-core/src/integrationTest/java/org/apache/geode/internal/cache/eviction/EvictionIntegrationTest.java]: There are a handful of places in the tests where, for instance, IntelliJ's inspections can identify where the old iterator style is used where the more readable `for each` loop style could be used instead.  A minor thing, but as long as the focus is refactoring, it's an easy thing to let your IDE do.

File: geode-core/src/distributedTest/java/org/apache/geode/internal/cache/eviction/EvictionDUnitTest.java [hunk 2]
```diff
@@ -1,209 +1,330 @@
-/*
- * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
- * agreements. See the NOTICE file distributed with this work for additional information regarding
- * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
```
Comment (dschneider-pivotal) [inline geode-core/src/distributedTest/java/org/apache/geode/internal/cache/eviction/EvictionDUnitTest.java]: Add back this copyright header

File: geode-core/src/integrationTest/java/org/apache/geode/internal/cache/eviction/EvictionIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,190 @@
+package org.apache.geode.internal.cache.eviction;
```
Comment (dschneider-pivotal) [inline geode-core/src/integrationTest/java/org/apache/geode/internal/cache/eviction/EvictionIntegrationTest.java]: Add a copyright header to this new file

**Comment by jinmeiliao** [comment]
The refactored tests are at most as flaky as the old one. Ran streeTest on old tests and get a lot more failures: https://github.com/apache/geode/pull/2745. I think we can call this refactor good.

**Pull request merged**